### PR TITLE
added KeebMonkey KBMG68.

### PR DESCRIPTION
**Pull request by ymlmkb**
Added new keyboard KBMG68.  Also known as FK680.  Retailer: KeebMonkey.  Manuf: unknown  

## Description

I am a consumer.  I have no affiliation with KeebMonkey or the manufacturer of this keeb, whoever that is.  Apologies in advance if it is improper for me to submit this PR for a keeb I didn't create.  I'm just trying to help out.  If this was not the right way then please let me know what is the right way, if you could, please.

I bought this keeb because QMK compatibility was advertised.  However there was no obvious subdir in the QMK repo yet.

So with help from folks on the QMK discord I was able to take the hex file I had found here:
  https://drive.google.com/file/d/1lSjk0ttjgK2JJ47j7fhvrnRjOUfXj9uY/view?usp=sharing
and run it through here:
  https://noroadsleft.github.io/kbf_qmk_converter/
and come up with mostly-working QMK code.  I tweaked it a little and successfully flashed my board.  All the keys work and RGB works.  This is the extent of my testing.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

None.

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keebmonkey/kbmg68/config.h [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+// /*== or choose animations ==*/
+//     #define RGBLIGHT_EFFECT_BREATHING
+//     #define RGBLIGHT_EFFECT_RAINBOW_MOOD
+//     #define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+//     #define RGBLIGHT_EFFECT_SNAKE
+//     #define RGBLIGHT_EFFECT_KNIGHT
+//     #define RGBLIGHT_EFFECT_CHRISTMAS
+//     #define RGBLIGHT_EFFECT_STATIC_GRADIENT
+//     #define RGBLIGHT_EFFECT_RGB_TEST
+//     #define RGBLIGHT_EFFECT_ALTERNATING
```
Comment (drashna) [inline keyboards/keebmonkey/kbmg68/config.h:66]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by fauxpark — CHANGES_REQUESTED** [review]
Missing a readme.

File: keyboards/keebmonkey/kbmg68/readme.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# KBMG68
+
+![KBMG68](https://www.keebmonkey.com/products/kbmg68)
```
Comment (fauxpark) [inline keyboards/keebmonkey/kbmg68/readme.md:3]: Needs to be an actual image link.

File: keyboards/keebmonkey/kbmg68/readme.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+The Keebmonkey KBMG68 should be the cheapest gasket keyboard you can ever have. Both the plate and the PCB are gasket mounted, resulting in a deep thocky sound. The shell of the kit is made of 6-layers of acrylic boards, leaving huge space for modding and continuous improvements (even though this kit already sounds perfect when it's out of the box).
+
+Keyboard Maintainer: Unknown
+Hardware Supported: Unknown
+Hardware Availability: [Keebmonkey KBMG68 Double Mounted Gasket Keyboard Kit](https://www.keebmonkey.com/products/kbmg68)
```
Comment (fauxpark) [inline keyboards/keebmonkey/kbmg68/readme.md:9]:
```suggestion
* Keyboard Maintainer: Unknown
* Hardware Supported: Unknown
* Hardware Availability: [Keebmonkey KBMG68 Double Mounted Gasket Keyboard Kit](https://www.keebmonkey.com/products/kbmg68)
```

File: keyboards/keebmonkey/kbmg68/kbmg68.h [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+#include "quantum.h"
+
+#define XXX KC_NO
+
+#define LAYOUT( \
+    K00, K01, K02, K03, K04, K05, K06, K07, K08, K09, K0A, K0B, K0C, K0E, K0F, \
+    K10, K12, K13, K14, K15, K16, K17, K18, K19, K1A, K1B, K1C, K1D, K1E, K1F, \
+    K20, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B, K2C, K2D, K2F, \
+    K31, K32, K33, K34, K35, K36, K37, K38, K39, K3A, K3B, K3D, K3E, K3F, \
+    K40, K41, K43, K46, K4A, K4B, K4C, K4D, K4E, K4F  \
```
Comment (fauxpark) [inline keyboards/keebmonkey/kbmg68/kbmg68.h:28]:
```suggestion
#define LAYOUT_65_ansi( \
    K00, K01, K02, K03, K04, K05, K06, K07, K08, K09, K0A, K0B, K0C, K0E, K0F, \
    K10, K12, K13, K14, K15, K16, K17, K18, K19, K1A, K1B, K1C, K1D, K1E, K1F, \
    K20, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B, K2C,      K2D, K2F, \
    K31,      K32, K33, K34, K35, K36, K37, K38, K39, K3A, K3B, K3D, K3E, K3F, \
    K40, K41, K43,                K46,                K4A, K4B, K4C, K4D, K4E, K4F  \
```

File: keyboards/keebmonkey/kbmg68/info.json [hunk 5]
```diff
@@ -0,0 +1,80 @@
+{
+    "keyboard_name": "kbmg68",
+    "url": "",
+    "maintainer": "qmk",
+    "layouts": {
+        "LAYOUT": {
```
Comment (fauxpark) [inline keyboards/keebmonkey/kbmg68/info.json:6]:
```suggestion
        "LAYOUT_65_ansi": {
```

File: keyboards/keebmonkey/kbmg68/rules.mk [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+NKRO_ENABLE = no            # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+
+# generated by KBFirmware JSON to QMK Parser
+# https://noroadsleft.github.io/kbf_qmk_converter/
```
Comment (fauxpark) [inline keyboards/keebmonkey/kbmg68/rules.mk:22]:
```suggestion
AUDIO_ENABLE = no           # Audio output

LAYOUTS = 65_ansi
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/keebmonkey/kbmg68/readme.md (hunk 2 above)
Comment (ymlmkb) [reply keyboards/keebmonkey/kbmg68/readme.md:3]: i don't think you understand LOL...this is barely even a product.  but ok.  i'll give a link to the absolutely sh*tty picture they show.

Comment (fauxpark) [reply keyboards/keebmonkey/kbmg68/readme.md:3]: `` is Markdown syntax for an image. The URL you've provided is invalid for that.

Comment (ymlmkb) [reply keyboards/keebmonkey/kbmg68/readme.md]: Ah ok it is I who did not understand.  Well anyway, i have updated the md...this is the best image they have and it sucks bro.  Sorry :-|

File: keyboards/keebmonkey/kbmg68/info.json [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+                { "label": "K4A (B5,C6)", "x": 10, "y": 4 },
+                { "label": "K4B (B5,C7)", "x": 11, "y": 4 },
+                { "label": "K4C (B5,F7)", "x": 12, "y": 4 },
+                { "label": "K4D (B5,F6)", "x": 13, "y": 4 },
+                { "label": "K4E (B5,F5)", "x": 14, "y": 4 },
+                { "label": "K4F (B5,F4)", "x": 15, "y": 4 }
+            ]
+        }
+    }
+    ,"meta": "https://noroadsleft.github.io/kbf_qmk_converter/"
```
Comment (noroadsleft) [inline keyboards/keebmonkey/kbmg68/info.json:79]:
```suggestion
```

File: keyboards/keebmonkey/kbmg68/kbmg68.c [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "kbmg68.h"
+
+// generated by KBFirmware JSON to QMK Parser
+// https://noroadsleft.github.io/kbf_qmk_converter/
```
Comment (noroadsleft) [inline keyboards/keebmonkey/kbmg68/kbmg68.c:20]:
```suggestion
```

File: keyboards/keebmonkey/kbmg68/kbmg68.h [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+) { \
+    { K00, K01, K02, K03, K04, K05, K06, K07, K08, K09, K0A, K0B, K0C, XXX, K0E, K0F }, \
+    { K10, XXX, K12, K13, K14, K15, K16, K17, K18, K19, K1A, K1B, K1C, K1D, K1E, K1F }, \
+    { K20, XXX, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B, K2C, K2D, XXX, K2F }, \
+    { XXX, K31, K32, K33, K34, K35, K36, K37, K38, K39, K3A, K3B, XXX, K3D, K3E, K3F }, \
+    { K40, K41, XXX, K43, XXX, XXX, K46, XXX, XXX, XXX, K4A, K4B, K4C, K4D, K4E, K4F }, \
+}
+
+// generated by KBFirmware JSON to QMK Parser
+// https://noroadsleft.github.io/kbf_qmk_converter/
```
Comment (noroadsleft) [inline keyboards/keebmonkey/kbmg68/kbmg68.h:38]:
```suggestion
```

File: keyboards/keebmonkey/kbmg68/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+    [0] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/keebmonkey/kbmg68/keymaps/default/keymap.c:21]:
```suggestion
    [0] = LAYOUT_65_ansi(
```

File: keyboards/keebmonkey/kbmg68/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_GRV,  
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,  
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,  KC_PGUP, 
+        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,   KC_PGDN, 
+        KC_LCTL, KC_LGUI, KC_LALT, KC_SPC,  KC_RALT, MO(1),   KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT  
+    ),
+
+    [1] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/keebmonkey/kbmg68/keymaps/default/keymap.c:29]:
```suggestion
    [1] = LAYOUT_65_ansi(
```

File: keyboards/keebmonkey/kbmg68/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+        RGB_TOG, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  _______, RGB_MOD, 
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, 
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, 
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, RGB_VAI, _______, 
+        MO(2),   _______, _______, _______, _______, _______, _______, _______, RGB_VAD, _______  
+    ),
+
+    [2] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/keebmonkey/kbmg68/keymaps/default/keymap.c:37]:
```suggestion
    [2] = LAYOUT_65_ansi(
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/keebmonkey/kbmg68/config.h [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+#define RGBLIGHT_EFFECT_KNIGHT
+#define RGBLIGHT_EFFECT_CHRISTMAS
+#define RGBLIGHT_EFFECT_STATIC_GRADIENT
+#define RGBLIGHT_EFFECT_RGB_TEST
+#define RGBLIGHT_EFFECT_ALTERNATING
+#define RGBLIGHT_EFFECT_TWINKLE
+#endif
+
+// generated by KBFirmware JSON to QMK Parser
+// https://noroadsleft.github.io/kbf_qmk_converter/
```
[the remark this answers is not in the document]
Comment (ymlmkb) [reply keyboards/keebmonkey/kbmg68/config.h:68]: These comments indicate how this file was generated.  I'm giving credit where credit is due.

File: keyboards/keebmonkey/kbmg68/info.json (hunk 7 above)
Comment (ymlmkb) [reply keyboards/keebmonkey/kbmg68/info.json:79]: These comments indicate how this file was generated.  I'm giving credit where credit is due.

File: keyboards/keebmonkey/kbmg68/kbmg68.h (hunk 9 above)
Comment (ymlmkb) [reply keyboards/keebmonkey/kbmg68/kbmg68.h:38]: These comments indicate how this file was generated.  I'm giving credit where credit is due.

File: keyboards/keebmonkey/kbmg68/readme.md [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+* Hardware Supported: Unknown
+* Hardware Availability: [Keebmonkey KBMG68 Double Mounted Gasket Keyboard Kit](https://www.keebmonkey.com/products/kbmg68)
+
+## Instructions
+
+### Build
+
+Make example for this keyboard (after setting up your build environment):
+
+    make keebmonkey/kbmg68:default
```
Comment (fauxpark) [inline keyboards/keebmonkey/kbmg68/readme.md:17]:
```suggestion
    make keebmonkey/kbmg68:default

Flashing example for this keyboard:

    make keebmonkey/kbmg68:default:flash
```

File: keyboards/keebmonkey/kbmg68/readme.md [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+### Flash
+
+- Unplug
+- Hold Escape
+- Plug In
+- Release Escape
+- `make keebmonkey/kbmg68:default:flash`
```
Comment (fauxpark) [inline keyboards/keebmonkey/kbmg68/readme.md:27]:
```suggestion
### Reset to Bootloader

Hold the Escape key while plugging the board in.
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**